api/apiFunctions.ts

```diff
@@ -40,38 +73,58 @@ export async function getAllWeeklyPicks() {
     return response;
   } catch (error) {
     console.error(error);
+    throw new Error('Error getting all weekly picks');
   }
 }
 
-export async function getNFLTeams() {
+/**
+ * Get all NFL teams
+ *
+ * @return {Models.DocumentList<Models.Document>} - The session object or an error
+ */
+export async function getNFLTeams(): Promise<
+  Models.DocumentList<Models.Document>
+> {
   try {
     return await databases.listDocuments(
       appwriteConfig.databaseId,
       '662152bfabacfbda3bb3',
     );
   } catch (error) {
     console.error(error);
+    throw new Error('Error getting NFL teams');
   }
 }
 
-export async function registerAccount(request: {
-  email: string;
-  password: string;
-}) {
-  const { email, password } = request;
-
+/**
+ * Register a new account
+ *
+ * @return {Models.User<Models.Preferences> | Error} - The user object or an error
+ */
+export async function registerAccount({
+  email,
+  password,
+}: IAccountData): Promise<Models.User<Models.Preferences>> {
   try {
     return await account.create(ID.unique(), email, password);
-  } catch (error: any) {
-    throw error;
+  } catch (error) {
+    console.error(error);
```

Comment:
On `register -> page.tsx` file, it `console.error('Registration Failed', error);`. Would this be redundant?